### ISSUE TITLE
Lethal Enforcers does not register shots

### ISSUE THREAD

**Issue by rtretiakov** [bsnes-emu/bsnes#75]
This could be two separate issues, but I suspect they are related.

When playing Lethal Enforcers, it seems impossible to get past the target "calibration" screen. It's almost as if something is "stuck". It could be that I am simply doing something wrong, but when I tried implementing this feature in libretro I could not get it working there either.

Another thing to add is that when setting up input, simply clicking the mouse twice when trying to assign the trigger will cause a segmentation fault - in other words, you must click the options at the bottom of the window for one of the three mouse buttons, but nowhere else in the window or you will segfault. This is the case when there is a game playing, or when there is no game playing. It also does not seem to be affected by what the controller port 2 device is set to in the main window.

OS: Linux (maybe others?)

**Comment by Screwtapello** [comment]
I tried to set up a Justifier, clicked about, and crashed bsnes. I rebuilt in debug mode, tried again, and couldn't reproduce the crash. Then I rebuilt *outside* of debug mode, and still couldn't reproduce it. I don't doubt there's crashes in that code, I'm sure it's been ticklish before, but I can't reproduce it now.

I can also confirm that Lethal Enforcers doesn't seem to work.

- Configure the bindings for the Justifier in port 2, mapping X and Y to the mouse axes and Trigger and Start to the buttons
- Configure the "Toggle Mouse Capture" hotkey
- Connect the Justifier to port 2
- Load up *Lethal Enforcers*
- I think if it detects a Justifier connected, it takes you straight to the "calibration" screen, with a target, a blinking "shoot at screen" message, and messages in the top left and top right saying "aim at screen" for player 1 and player 2
- There's a green crosshair cursor on the screen, if you press the Toggle Mouse Capture button you can control it more reliably
- Pressing the Trigger binding makes the screen flash white
- The game never seems to accept that as valid input, and the messages in the top-left and top-right never change.

**Comment by Screwtapello** [comment]
"De andere Max" on Discord figured out the crash occurs when you click below the last item in the table, while assigning an input. From that it was easy enough to get a backtrace and find the problem. Thanks for reporting it!

**Comment by Screwtapello** [comment]
I updated my previous comment to include proper verified steps to reproduce.

I used git bisect to discovver the regression occurred with commit fb95d5b59f2661b698987072c0545620ec686751, whose commit message mentions "Moved frame events into the CPU core to prevent PPU<>NMI race condition." The only other change in that commit is about the frame advance hotkey, so I'm pretty sure that change is what broke things. I have no idea *why* it broke things, though, or how to fix it without re-introducing the mentioned race condition.
